server/src/main/java/io/seata/server/transaction/saga/SagaCore.java

```diff
@@ -194,12 +194,12 @@ public boolean doGlobalRollback(GlobalSession globalSession, boolean retrying) t
     public void doGlobalReport(GlobalSession globalSession, String xid, GlobalStatus globalStatus) throws TransactionException {
         if (GlobalStatus.Committed.equals(globalStatus)) {
             SessionHelper.removeAllBranch(globalSession, false);
-            SessionHelper.endCommitted(globalSession,false);
+            SessionHelper.endSagaGlobalSession(globalSession, GlobalStatus.Committing);
             LOGGER.info("Global[{}] committed", globalSession.getXid());
         } else if (GlobalStatus.Rollbacked.equals(globalStatus)
                 || GlobalStatus.Finished.equals(globalStatus)) {
             SessionHelper.removeAllBranch(globalSession, false);
-            SessionHelper.endRollbacked(globalSession, false);
+            SessionHelper.endSagaGlobalSession(globalSession, GlobalStatus.Rollbacking);
```

Comment:
same above.

Reply:
> same above.

I decide to end the GlobalSession when tc report Committed or Rollbacked